source/MRCuda/MRCudaMeshProject.h

```diff
@@ -1,10 +1,24 @@
 #pragma once
 #include "exports.h"
-#include "MRMesh/MRMesh.h"
+#include "MRMesh/MRPointsToMeshProjector.h"
 
 namespace MR { namespace Cuda {
 
-/// Computes distance of 2d contours according to ContourToDistanceMapParams (works correctly only when withSign==false)
-MRCUDA_API std::vector<MR::MeshProjectionResult> findProjections( const std::vector<Vector3f>& points, const MR::Mesh& mesh, float upDistLimitSq = FLT_MAX, float loDistLimitSq = 0);
+struct MeshProjectorData;
+/// Computes the closest point on mesh to each of given points on GPU
+class PointsToMeshProjector : public IPointsToMeshProjector
+{
+    std::shared_ptr<MeshProjectorData> meshData_;
+
+public:
+    MRCUDA_API PointsToMeshProjector();
+    /// update transforms applied to the points and to the referencing mesh
+    MRCUDA_API virtual void updateTransforms( const AffineXf3f& objXf, const AffineXf3f& refObjXf ) override;
+    /// update all data related to the referencing mesh
+    MRCUDA_API virtual void updateMeshData( std::shared_ptr<const Mesh> mesh ) override;
+    /// Computes the closest point on mesh to each of given points
+    MRCUDA_API virtual std::vector<MR::MeshProjectionResult> findProjections( const std::vector<Vector3f>& points, float upDistLimitSq, float loDistLimitSq ) override;
```

Comment:
mb do not `return std::vector<MR::MeshProjectionResult>`  but take 
`std::vector<MR::MeshProjectionResult>& result` argumnt (to be able to avoid CPU allocations each call)